### PR TITLE
Expand Cypress targeting tests

### PR DESCRIPTION
**Pull request by domlander**
## What does this change?

- (functional) Expand targeting tests.
- (functional) Add a timeout of 30secs to finding the Sourcepoint iframe: there were transient timeout failures with a 10sec timeout.
- Moved Prebid targeting into the targeting file.
- Use AllowAllConsent helper function everywhere cookies are accepted.
- Reorganise test page fixtures into their own files
- Sort Page targeting alphabetically

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [ ] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: cypress/e2e/consent.cy.ts [hunk 1]
```diff
@@ -67,12 +63,12 @@ const expectAdFree = (reasons: AdFreeCookieReasons[]) => {
...
-    beforeEach(() => {
-        cy.clearCookies();
-        cy.clearLocalStorage();
-    });
+	beforeEach(() => {
+		cy.clearCookies();
+		cy.clearLocalStorage();
+	});
 
-	pages.forEach(({ path, adTest }) => {
```
Comment (domlander) [inline cypress/e2e/consent.cy.ts:75]: Nothing has functionally changed, we still test one page: the home page. Before, there was only one front that existed and we set `pages` to be this front. Now, we use the first item in the array of fronts.

Comment (domlander) [reply cypress/e2e/consent.cy.ts:75]: We only want to test one page and this suite of tests take a long time and we would not get much more value from testing extra pages.

**Review by arelra — APPROVED** [review]
Great testing improvements and refactoring!

File: cypress/e2e/targeting.cy.ts [hunk 2]
```diff
@@ -1,38 +1,132 @@
...
+	});
+});
 
-		// Click "Yes, I'm happy" on the sourcepoint banner to obtain consent
-		cy.getIframeBody('sp_message_iframe_').find('.btn-primary').click();
+describe('Prebid targeting', () => {
+	const interceptGamRequest = () =>
+		cy.intercept(
+			{
+				url: 'https://securepubads.g.doubleclick.net/gampad/ads?**',
```
[the remark this answers is not in the document]
Comment (domlander) [reply cypress/e2e/targeting.cy.ts:86]: Yep, good catch

File: cypress/fixtures/pages/articles.ts [hunk 3]
```diff
@@ -1,15 +1,9 @@
-import { getStage, getTestUrl } from '../lib/util';
+import { getStage, getTestUrl } from '../../lib/util';
+import { Page } from './Page';
```
[the remark this answers is not in the document]
Comment (domlander) [reply cypress/fixtures/pages/articles.ts:2]: Yeah it should be

[the remark this answers is not in the document]
Comment (Jakeii) [reply cypress/fixtures/pages/articles.ts:2]: Maybe add a rule to the cypress eslintrc? or make it extend the main one.

File: cypress/e2e/merchandising.cy.ts [hunk 4]
```diff
@@ -5,21 +5,24 @@ describe('merchandising slot on pages', () => {
...
 				// creative isn't loaded unless slot is in view
-				cy.get('#dfp-ad--merchandising').scrollIntoView({duration: 4000});
+				cy.get('#dfp-ad--merchandising').scrollIntoView({
+					duration: 4000,
+				});
 
 				// Check that an iframe is placed inside the merchandising ad slot
-				cy.get('#dfp-ad--merchandising').find('iframe', {timeout: 10000}).should('exist');
+				cy.get('#dfp-ad--merchandising')
+					.find('iframe', { timeout: 30000 })
```
[the remark this answers is not in the document]
Comment (domlander) [reply cypress/e2e/merchandising.cy.ts]: I have added this, including the timeout of 30s. If we want to customise this in the future for certain ads, then we can add a parameter.

**Review by arelra — APPROVED** [review]
🎉

**Pull request merged**